iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LogicalPlanBuilder.java

```diff
@@ -301,6 +313,54 @@ public LogicalPlanBuilder planLast(
       }
     }
 
+    if (lastQueryNonWriteViewExpressions != null) {
+      for (Expression expression : lastQueryNonWriteViewExpressions) {
+        Set<Expression> sourceTransformExpressions =
+            new HashSet<>(Collections.singletonList(expression));
+        LogicalPlanBuilder planBuilder = new LogicalPlanBuilder(analysis, context);
+        Set<Expression> sources = new LinkedHashSet<>(searchSourceExpressions(expression));
+        FunctionExpression maxTimeAgg =
+            new FunctionExpression(
+                MAX_TIME, new LinkedHashMap<>(), Collections.singletonList(expression));
+        FunctionExpression lastValueAgg =
+            new FunctionExpression(
+                LAST_VALUE, new LinkedHashMap<>(), Collections.singletonList(expression));
+        analyzeExpression(analysis, expression);
+        analyzeExpression(analysis, maxTimeAgg);
+        analyzeExpression(analysis, lastValueAgg);
+
+        planBuilder =
+            planBuilder
+                .planRawDataSource(
+                    sources,
+                    queryStatement.getResultTimeOrder(),
+                    analysis.getGlobalTimeFilter(),
+                    analysis.isLastLevelUseWildcard())
+                .planWhereAndSourceTransform(
+                    null,
+                    sourceTransformExpressions,
+                    false,
+                    queryStatement.getSelectComponent().getZoneId(),
+                    queryStatement.getResultTimeOrder())
+                .planAggregation(
+                    new LinkedHashSet<>(Arrays.asList(maxTimeAgg, lastValueAgg)),
+                    null,
+                    analysis.getGroupByTimeParameter(),
+                    analysis.getGroupByParameter(),
+                    false,
+                    AggregationStep.SINGLE,
+                    Ordering.ASC);
+
+        LastQueryTransformNode transformNode =
+            new LastQueryTransformNode(
+                context.getQueryId().genPlanNodeId(),
+                Arrays.asList(planBuilder.getRoot()),
```

Comment:
```suggestion
                Collections.singletonList(planBuilder.getRoot()),
```

Reply:
Using Collections.singletonList may meets error when calling "clone" method of PlanNode, I will retry again.